packages/pages/src/components/analytics/hooks.ts

```diff
@@ -0,0 +1,96 @@
+import { ConversionDetails, Visitor } from "@yext/analytics";
+import { useContext } from "react";
+import { AnalyticsContext } from "./context";
+import { concatScopes } from "./helpers";
+import { AnalyticsMethods } from "./interfaces";
+import { useScope } from "./scope";
+
+declare global {
+  interface Window {
+    setAnalyticsOptIn: () => void;
+  }
+}
+
+/**
+ * The useAnalytics hook can be used anywhere in the tree below a configured
+ * AnalyticsProvider.  Calling it will return an object to give you access to
+ * the analytics convenience methods for use in your components,
+ * such as track(), pageView(), optIn() etc.
+ *
+ * @public
+ */
+export function useAnalytics(): AnalyticsMethods {
+  const ctx = useContext(AnalyticsContext);
+  if (!ctx) {
+    throw new Error(
+      "Attempted to call useAnalytics outside of an AnalyticsProvider"
+    );
+  }
+
+  // TODO: is this the right way / place to expose a callback for use by a Cookie Management banner?
+  if (!window.setAnalyticsOptIn) {
+    window.setAnalyticsOptIn = async () => {
+      await ctx.optIn();
+    };
+  }
+
+  const scope = useScope();
+
+  // TODO: this is ugly, I imagine there is a more elegant way of doing this
+  return {
+    trackClick(
+      eventName: string,
+      conversionData?: ConversionDetails
+    ): (e: MouseEvent) => Promise<void> {
+      return ctx.trackClick(eventName, conversionData);
```

Comment:
think we need to concatScopes here

Reply:
do we? trackClick calls track under the hood and the concatScopes call is there

Reply:
oh yeah you're right, nevermind